javav2/example_code/s3/src/main/java/com/example/s3/replication/BucketReplication.java

```diff
@@ -0,0 +1,411 @@
+            the following steps:
+                        
+            The application starts by initializing the necessary AWS clients, such as S3Client and 
+            IamClient, to interact with the AWS services. It then creates two S3 buckets, 
+            one as the source bucket and the other as the destination bucket. Next, it 
+            assigns an IAM role with the necessary permissions to access the source and destination 
+            buckets. 
+            
+            The application then enables bucket versioning for both the source and destination buckets, 
+            which is a prerequisite for enabling CRR. Finally, the application sets the replication 
+            configuration for the source bucket, specifying the destination bucket, the IAM role, 
+            and the replication rules.
+            
+            Lets get started...
+                                 
+            """);
+
+        waitForInputToContinue(scanner);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("1. Create a source and destination bucket");
+        waitForInputToContinue(scanner);
+        createBucket(s3Client, sourceBucketName);
+        createBucket(s3Client, destBucketName);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("2. Assign a role");
+        waitForInputToContinue(scanner);
+        assignRole(iamClient, roleName, sourceBucketName, destBucketName);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("3. Enable bucket versioning");
+        waitForInputToContinue(scanner);
+        enableBucketVersioning(s3Client, sourceBucketName);
+        enableBucketVersioning(s3Client, destBucketName);
+        System.out.println(DASHES);
+
+        System.out.println("4. Set replication on both buckets");
+        waitForInputToContinue(scanner);
+        setReplication(s3Client, sourceBucketName, destBucketName, destinationBucketARN, roleARN);
+        System.out.println(DASHES);
+
+        System.out.println("5. Get replication information");
+        waitForInputToContinue(scanner);
+        getReplicationDetails(s3Client, sourceBucketName);
+        System.out.println(DASHES);
+
+        System.out.println("6. Delete the AWS resources");
+        waitForInputToContinue(scanner);
+        deleteBucket(s3Client, sourceBucketName);
+        deleteBucket(s3Client, destBucketName);
+        deleteRole(iamClient, roleName);
+        System.out.println("This concludes the Cross-Region Replication example");
+        System.out.println(DASHES);
+    }
+
+    /**
+     * Enables bucket versioning for the specified S3 bucket.
+     *
+     * @param s3Client the S3 client to use for the operation
+     * @param bucketName the name of the S3 bucket to enable versioning for
+     */
+    public static void enableBucketVersioning(S3Client s3Client, String bucketName){
```

Comment:
Add PutBucketVersioning to the metadata? There are only CLI and Powershell examples.